src/Controller/DeveloperAppViewController.php

```diff
@@ -23,22 +23,61 @@
 use Drupal\apigee_edge\Entity\DeveloperStatusCheckTrait;
 use Drupal\Core\Entity\Controller\EntityViewController;
 use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Entity\EntityManagerInterface;
+use Drupal\Core\Messenger\MessengerInterface;
 use Drupal\Core\Render\Markup;
+use Drupal\Core\Render\RendererInterface;
 use Drupal\Core\Routing\RouteMatchInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
 
 /**
  * Displays the view page of a developer app on the UI.
  */
 class DeveloperAppViewController extends EntityViewController implements DeveloperAppPageTitleInterface {
 
   use DeveloperStatusCheckTrait;
+  use DeveloperAppCallbackUrlCheckTrait;
+
+  /**
+   * The messenger.
+   *
+   * @var \Drupal\Core\Messenger\MessengerInterface
+   */
+  protected $messenger;
+
+  /**
+   * Creates an EntityViewController object.
+   *
+   * @param \Drupal\Core\Entity\EntityManagerInterface $entity_manager
+   *   The entity manager.
+   * @param \Drupal\Core\Render\RendererInterface $renderer
+   *   The renderer service.
+   * @param \Drupal\Core\Messenger\MessengerInterface $messenger
+   *   The messenger.
+   */
+  public function __construct(EntityManagerInterface $entity_manager, RendererInterface $renderer, MessengerInterface $messenger) {
+    parent::__construct($entity_manager, $renderer);
+    $this->messenger = $messenger;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('entity.manager'),
+      $container->get('renderer'),
+      $container->get('messenger')
+    );
+  }
 
   /**
    * {@inheritdoc}
    */
   public function view(EntityInterface $developer_app, $view_mode = 'full') {
     /** @var \Drupal\apigee_edge\Entity\DeveloperAppInterface $developer_app */
     $this->checkDeveloperStatus($developer_app->getOwnerId());
+    $this->checkCallbackUrl($developer_app);
```

Comment:
This should get the view mode as a parameter and load the view mode instead of "default" always, isn't it?